types/adapters.d.ts

```diff
@@ -1,6 +1,14 @@
-import { InternalOptions } from "./internals"
-import { User, Profile, Session } from "."
+import { User, Session } from "."
 import { EmailConfig } from "./providers"
+import { Awaitable } from "./internals/utils"
+
+export interface AdapterUser extends User {
+  emailVerified?: Date | null
+}
+
+export interface AdapterSession extends Omit<Session, "expires"> {
+  expires: Date
+}
```

Comment:
`Session["expires"]` is a string by default so it can easily be passed to the client, but it makes more sense to retrieve it as `Date` from DB.

See: https://github.com/vercel/next.js/discussions/11498